### PR TITLE
Update SecurityProviderInterface.adoc to use Jakarta terms

### PR DESCRIPTION
**Pull request by xstefank**
Signed-off-by: xstefank <xstefank122@gmail.com>

### CODE REVIEW

File: specification/src/main/asciidoc/platform/ServiceProviderInterface.adoc [hunk 1]
```diff
@@ -1,64 +1,62 @@
...
-some or all of these facilities, as described in
-link:#a3212[See Profiles]”.
+The Jakarta™ EE Platform includes several technologies that are primarily
+intended to be used to extend the capabilities of the Jakarta EE containers.
+In addition, some Jakarta EE technologies include service provider interfaces
+along with their application programming interfaces. A Jakarta EE profile may
+include some or all of these facilities, as described in link:#a3212[See Profiles]”.
 
-=== Java™ EE Connector Architecture
+=== Jakarta™ EE Connectors
```
Comment (kazumura) [inline specification/src/main/asciidoc/platform/ServiceProviderInterface.adoc]: should be  Jakarta™ Connectors

Comment (xstefank) [reply specification/src/main/asciidoc/platform/ServiceProviderInterface.adoc]: Thanks. Fixed.

File: specification/src/main/asciidoc/platform/ServiceProviderInterface.adoc [hunk 2]
```diff
@@ -1,64 +1,62 @@
...
-=== Java™ Authorization Service Provider Contract for Containers
+=== Jakarta™ Authorization
 
-The JACC specification defines the contract
-between a Java EE container and an authorization policy provider.
+The Jakarta Authorization specification defines the contract
+between a Jakarta EE container and an authorization policy provider.
 
-The JACC specification is available at
+The Jakarta Authorization specification is available at
```
Comment (kwsutter) [inline specification/src/main/asciidoc/platform/ServiceProviderInterface.adoc]: This link is actually for the old JACC specification...  Since we don't have the updated Jakarta Authorization Spec yet, we should probably revert this change to explain that the original JACC spec is available here...  Either that or just remove this reference in this portion of the doc.

File: specification/src/main/asciidoc/platform/ServiceProviderInterface.adoc [hunk 3]
```diff
@@ -1,64 +1,62 @@
...
-managers, resource adapters, as well as EJB and Web application
+managers, resource adapters, as well as Jakarta Enterprise Beans and Web application
 components. This provides the ability to register synchronization
 objects with special ordering semantics, associate resource objects with
 the current transaction, get the transaction context of the current
 transaction, get current transaction status, and mark the current
 transaction for rollback.
 
-The JTA specification is available at
+The Jakarta Transactions specification is available at
```
Comment (kwsutter) [inline specification/src/main/asciidoc/platform/ServiceProviderInterface.adoc]: Same comment here...  This link is the Java EE Transactions doc, not the Jakarta Spec...  We should have a single reference section at the end of the doc with references to the old and new Specs.  Maybe we should just be pointing at that?  Or, just remove this explicit reference.

File: specification/src/main/asciidoc/platform/ServiceProviderInterface.adoc [hunk 4]
```diff
@@ -1,64 +1,62 @@
...
+=== Jakarta™ Persistence
 
-Java Persistence provides interfaces in the
+Jakarta Persistence provides interfaces in the
 _javax.persistence.spi_ package that allow a persistence provider to be
-plugged into the Java Persistence framework.
+plugged into the Jakarta Persistence framework.
 
-The Java Persistence specification is available
+The Jakarta Persistence specification is available
```
Comment (kwsutter) [inline specification/src/main/asciidoc/platform/ServiceProviderInterface.adoc]: Same comment again...

File: specification/src/main/asciidoc/platform/ServiceProviderInterface.adoc [hunk 5]
```diff
@@ -1,64 +1,62 @@
...
-can be discovered and used through the JavaMail API. This allows the
-JavaMail API to be extended with support for new mail protocols and
+The Jakarta Mail specification describes how
+Jakarta Mail protocol providers can be packaged and distributed so that they
+can be discovered and used through the Jakarta Mail API. This allows the
+Jakarta Mail API to be extended with support for new mail protocols and
 mailbox formats.
 
-The JavaMail specification is available at
+The Jakarta Mail specification is available at
```
Comment (kwsutter) [inline specification/src/main/asciidoc/platform/ServiceProviderInterface.adoc]: And, again...

File: specification/src/main/asciidoc/platform/ServiceProviderInterface.adoc (hunk 3 above)
Comment (xstefank) [reply specification/src/main/asciidoc/platform/ServiceProviderInterface.adoc]: I agree. Can you point me to the new spec documents, please? I can't find them.

**Comment by xstefank** [comment]
@kwsutter I moved JCP links to the end of the documents and added section also for new Jakarta spec documents. I will make a note to update this section when the documents will be available.

**Comment by xstefank** [comment]
@kwsutter can you take a long again, please?

File: specification/src/main/asciidoc/platform/ServiceProviderInterface.adoc (hunk 3 above)
Comment (kwsutter) [reply specification/src/main/asciidoc/platform/ServiceProviderInterface.adoc]: We don't have the "standard" location for the updates specs yet...  At this point, let's just be consistent.  If we want to reference the JCP spec, then that's okay.  But, we shouldn't talk about Jakarta Spec and then point at the JCP location.

**Comment by kwsutter** [comment]
@xstefank Thanks, Martin.  That's a good compromise by moving the references to the end of the section.  If or when we have the proper specs for Jakarta, we can update this section.  Thanks.  I will approve and merge.

**Comment by xstefank** [comment]
Thanks @kwsutter!

File: specification/src/main/asciidoc/platform/ServiceProviderInterface.adoc [hunk 6]
```diff
@@ -1,64 +1,61 @@
...
+Jakarta Mail protocol providers can be packaged and distributed so that they
+can be discovered and used through the Jakarta Mail API. This allows the
+Jakarta Mail API to be extended with support for new mail protocols and
 mailbox formats.
 
-The JavaMail specification is available at
-_http://jcp.org/en/jsr/detail?id=919_ .
+=== Specification documents
+
+==== Original Java™ EE specification documents
```
Comment (bshannon) [inline specification/src/main/asciidoc/platform/ServiceProviderInterface.adoc]: It will be cleaner and less confusing if we don't reference the JCP specs at all.
It would be better to omit these sections at the end and just depend on the
appendix that will link to all the specification documents.

Comment (xstefank) [reply specification/src/main/asciidoc/platform/ServiceProviderInterface.adoc]: @bshannon ok, but I believe there are more sections that reference original JCP links. It would be better to do this in one PR for all documents after the new spec links are available.

Comment (bshannon) [reply specification/src/main/asciidoc/platform/ServiceProviderInterface.adoc]: The expected form of spec links is described in [this message from Wayne](https://www.eclipse.org/lists/jakarta.ee-spec/msg00035.html).

Comment (xstefank) [reply specification/src/main/asciidoc/platform/ServiceProviderInterface.adoc]: Would then make sense to update it right away? I saw the message after the PR was merged.
